S2VX.Game.Tests/CommandTests.cs

```diff
@@ -0,0 +1,150 @@
+
+        [Test]
+        public void ValueAt_Double_0DurationUsesStartValue() {
+            var inputCurrentTime = 1.0f;
+            var inputStartValue = 0.1f;
+            var inputEndValue = 1.0f;
+            var inputStartTime = 1.0f;
+            var inputEndTime = 1.0f;
+            var inputEasing = Easing.None;
+            var expected = inputStartValue;
+            var actual = Interpolation.ValueAt(inputCurrentTime, inputStartValue, inputEndValue, inputStartTime, inputEndTime, inputEasing);
+            var tolerance = 0.001;
+            Assert.AreEqual(expected, actual, tolerance);
+        }
+
+        [Test]
+        public void ValueAt_Color_0StartTime0DurationUsesStartValue() {
+            var inputCurrentTime = 1.0f;
+            var inputStartValue = Color4.White;
+            var inputEndValue = Color4.Black;
+            var inputStartTime = 0.0f;
+            var inputEndTime = 0.0f;
+            var inputEasing = Easing.None;
+            var expected = inputStartValue;
+            var actual = Interpolation.ValueAt(inputCurrentTime, inputStartValue, inputEndValue, inputStartTime, inputEndTime, inputEasing);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void ValueAt_Color_0DurationUsesStartValue() {
+            var inputCurrentTime = 1.0f;
+            var inputStartValue = Color4.White;
+            var inputEndValue = Color4.Black;
+            var inputStartTime = 1.0f;
+            var inputEndTime = 1.0f;
+            var inputEasing = Easing.None;
+            var expected = inputStartValue;
+            var actual = Interpolation.ValueAt(inputCurrentTime, inputStartValue, inputEndValue, inputStartTime, inputEndTime, inputEasing);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void ValueAt_Vector2_0StartTime0DurationUsesStartValue() {
+            var inputCurrentTime = 1.0f;
+            var inputStartValue = Vector2.Zero;
+            var inputEndValue = Vector2.One;
+            var inputStartTime = 0.0f;
+            var inputEndTime = 0.0f;
+            var inputEasing = Easing.None;
+            var expected = inputStartValue;
+            var actual = Interpolation.ValueAt(inputCurrentTime, inputStartValue, inputEndValue, inputStartTime, inputEndTime, inputEasing);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void ValueAt_Vector2_0DurationUsesStartValue() {
+            var inputCurrentTime = 1.0f;
+            var inputStartValue = Vector2.Zero;
+            var inputEndValue = Vector2.One;
+            var inputStartTime = 1.0f;
+            var inputEndTime = 1.0f;
+            var inputEasing = Easing.None;
+            var expected = inputStartValue;
+            var actual = Interpolation.ValueAt(inputCurrentTime, inputStartValue, inputEndValue, inputStartTime, inputEndTime, inputEasing);
+            Assert.AreEqual(expected, actual);
+        }
+
```

Comment:
Just to ensure my sanity, can we also add a few cases for great than 0 Duration? My assumption is that all of these should use the end value, but I'd like the tests to confirm this.